README.rst

```diff
@@ -60,37 +60,75 @@ Here's example of the things needed inside a Scrapy project's ``settings.py`` fi
 
     TWISTED_REACTOR = "twisted.internet.asyncioreactor.AsyncioSelectorReactor"
 
-How to use
-----------
+Usage
+-----
 
-Set the ``zyte_api`` `Request.meta
-<https://docs.scrapy.org/en/latest/topics/request-response.html#scrapy.http.Request.meta>`_
-key to download a request using Zyte API. Full list of parameters is provided in the
-`Zyte API Specification <https://docs.zyte.com/zyte-api/openapi.html#zyte-openapi-spec>`_.
+To enable every request to be sent through Zyte API, you can set the following
```

Comment:
I'm not sure the setting below makes every request to be sent through Zyte API. If I'm reading the code correctly, it first checks if zyte_api meta key is present and the value is true-ish, and uses ZYTE_API_DEFAULT_PARAMS only in this case. So, parameters are only applied for requests which are already marked explicitly as Zyte API requests.

That's actually the behavior I'm fine with :) It looks useful e.g. to set default geolocation for Zyte API requests made from a spider. On the other hand, making every request going through Zyte API by using this option looks quite problematic; it'd require more thought. For example,

* would this example mean that robots.txt files will be downloaded through Zyte API with browserHtml: True? 
* would it mean that e.g. a POST request won't work properly, while looking like a regular request in the spider code, and without raising any kind of warning? 

It seems some feature to "enable Zyte API transparently" makes sense after we put a bit more effort into "transparent" integration of scrapy Requests with Zyte API direct downloader, it doesn't look as straightforward as setting default parameters.

Reply:
> I'm not sure the setting below makes every request to be sent through Zyte API. If I'm reading the code correctly, it first checks if zyte_api meta key is present and the value is true-ish, and uses ZYTE_API_DEFAULT_PARAMS only in this case. So, parameters are only applied for requests which are already marked explicitly as Zyte API requests.

That's right. https://github.com/scrapy-plugins/scrapy-zyte-api/pull/13 attempted to not change the behavior fully. It only addressed to prevent the user from repeating the same Zyte API parameters in different parts of the code. In any case, hopefully it should serve as a base point to enable all requests go through Zyte API. :) 

> would this example mean that robots.txt files will be downloaded through Zyte API with browserHtml: True?

That's a good point. This also extends to other things like downloading an image, file, etc.

I think one option is to prevent `browserHtml` from being set in the `ZYTE_API_DEFAULT_PARAMS`. The same is true for `javascript`, `product`, `articles`, etc. I think one way to think of it is having `ZYTE_API_DEFAULT_PARAMS` support only a handful of parameters, like `geolocation` or `httpResponseBody` (though it needs to toggled off when `browserHtml` is turned on), etc.

This could be a bit tricky since Zyte API offers a lot of features that doesn't only cater to being a simply proxy. I think for users to have a seamless experience when using Zyte API, we should have an interface that could cover general Scrapy use. This could mean only downloading the `httpResponseBody` and perhaps `httpResponseHeaders` by default which is available to all `scrapy.Request` unless overridden by the `zyte_api` param in **meta**.

We can open up another PR to explore the different options here.

> would it mean that e.g. a POST request won't work properly, while looking like a regular request in the spider code, and without raising any kind of warning?

I think this could be alleviated somehow by the proposed interface in https://github.com/scrapy-plugins/scrapy-zyte-api/pull/20.

Reply:
I think @kmike’s main point was that the documentation needs an update, as it is currently quite misleading. The other points are about issues if we decided to make the implementation match the current documentation, which I don’t think we want, at least not at this point.

Reply:
I see, thanks for clarifying! What do you think about this doc update? https://github.com/scrapy-plugins/scrapy-zyte-api/pull/10/commits/2adc8a63b984847c693c41cc70d13794213eb237

Reply:
Hmm a good point. There could be a few cases for this:

1. The user is aware that `ZYTE_API_DEFAULT_PARAMS` is set in the settings and wish to use exactly the same values. Thus, `zyte_api` would be represented as an empty `{}`.
2.  `ZYTE_API_DEFAULT_PARAMS` is not in the settings. The user might've thought that it was set but was actually not. This would result in the requests to be executed without using Zyte API.

We could use `meta={"zyte_api": {}}` to represent it but it kinda feels unnatural as an empty-dict. Perhaps we could also support `meta={"zyte_api": True}` as another alternative. The `{}` is still a valid use-case though since the user might have a dynamic process to determine the contents of meta field. For example:
```python
meta= {"zyte_api": {}}  # default
if does_it_look_like_we_need_javascript():
    meta["zyte_api"].update({"javascript": True})
if how_about_using_fr_region():
    meta["zyte_api"].update({"geolocation": "FR"})
yield scrapy.Request(url, self.callback_func, meta=meta)
```

To summarize, we could use `{}` and `True` to enable Zyte API Requests bearing in mind the need for `ZYTE_API_DEFAULT_PARAMS` be set.

Any thoughts on this?

Reply:
+1 to having `{}` and `True` work and behave the same. At some point Zyte Data API may (should? :slightly_smiling_face:) work without parameter other than the URL (e.g. consider `browserHtml` or `httpResponseBody` enabled if no other output is enabled).

Reply:
This seems to be the only remaining fix to make, the PR looks good to me otherwise 👍

Reply:
Refactored the code to accept `True` and `{}` in https://github.com/scrapy-plugins/scrapy-zyte-api/pull/10/commits/f5a9bb074d4f76b8ef294a84edb8c4946919c298.